pkg/utils/secrets_test.go

```diff
@@ -27,3 +38,53 @@ func TestCheckSums(t *testing.T) {
 		})
 	}
 }
+
+func testSecret() *corev1.Secret {
+	sc := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testSecretName,
+			Namespace: testNamespace,
+		},
+		Data: map[string][]byte{
+			testDataKey: []byte(testDataValue),
+		},
+	}
+	return sc
+}
+
+func testSecretNoData() *corev1.Secret {
+	sc := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      testSecretName,
+			Namespace: testNamespace,
+		},
+		Data: map[string][]byte{},
+	}
+	return sc
+}
+
+func TestLoadSecretData(t *testing.T) {
+	t.Run("Secret with datakey", func(t *testing.T) {
```

Comment:
Instead of individual t.Run statements, please reformat this into a table-based test, similar to `TestCheckSums` above